apps/builder/app/builder/features/style-panel/sections/box-shadows/box-shadow-content.tsx

```diff
@@ -0,0 +1,54 @@
+import { useState } from "react";
+import type { InvalidValue, StyleProperty } from "@webstudio-is/css-data";
+import { TextArea, theme } from "@webstudio-is/design-system";
+import type { RenderCategoryProps } from "../../style-sections";
+import { parseBoxShadow } from "./box-shadow-parser";
+
+type IntermediateValue = {
+  type: "intermediate";
+  value: string;
+};
+
+const property: StyleProperty = "boxShadow";
+
+export const BoxShadowContent = (
+  props: Pick<
+    RenderCategoryProps,
+    "deleteProperty" | "currentStyle" | "setProperty"
+  >
+) => {
+  const [intermediateValue, setIntermediateValue] = useState<
+    IntermediateValue | InvalidValue | undefined
+  >(undefined);
+  const styleInfo = props.currentStyle[property];
+  const styleValue = styleInfo?.value;
+
+  const textAreaValue =
+    intermediateValue?.value ??
+    (styleValue?.type === "unparsed" ? styleValue.value : undefined);
+
+  const handleChange = (value: string) => {
+    setIntermediateValue({
+      type: "intermediate",
+      value,
+    });
+
+    const shadows = parseBoxShadow(value);
+
+    if (shadows.type === "invalid") {
+      return;
```

Comment:
we also need to set invalid state like in other css values (red border, red text)